### PR TITLE
enhanced documentation

### LINKED ISSUE #1: Update README
### Update README to provide a clear and concise description of the repository.

<br>

Optional:

- Add hyperlinks
- Reorganize the repository
- Add [folder structure](https://github.com/kriasoft/Folder-Structure-Conventions/blob/master/README.md#folder-structure-conventions)

### PR DESCRIPTION
**Pull request by finn-str**
fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,73 @@
...
+  
+- **Variety of Problems**: Our repository covers a broad spectrum of problems, from the basics to the most challenging ones. Whether you're a novice or an expert, there's something here for everyone.
+  
+- **Clean and Commented Code**: We believe that the best code is the one that speaks for itself. Our solutions are written in a clean and readable manner, peppered with comments to guide you through.
+
+## Getting Started
+
+1. **Clone the Repository**: 
+   \```bash
+   git clone https://github.com/finn-str/Algorithm-Solutions.git
```
Comment (Alperencode) [inline README.md]: This link needs to be directed to my repository, like this:
`git clone https://github.com/Alperencode/Algorithm-Solutions.git`

File: README.md [hunk 2]
```diff
@@ -1,9 +1,73 @@
...
+│       ├── solution.java
+│       └── README.md
+│
+└── advanced/
+    └── problem1/
+        ├── solution.cpp
+        └── README.md
+```
+
+> Note: This is a generic folder structure. Replace with the actual structure of the repository.
```
Comment (Alperencode) [inline README.md:60]: This structure needs to be replaced with the structure of the repository, I can fix that later so not a problem for now.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,73 @@
...
+
+## Contribution Guidelines
+
+We're always on the lookout for better solutions and optimizations. If you believe you have a more efficient solution or a new algorithmic problem that could benefit the community:
+
+1. Fork the repository.
+2. Make your changes.
+3. Submit a [pull request](https://github.com/finn-str/Algorithm-Solutions/pulls) with a detailed explanation of your solution or problem.
```
Comment (Alperencode) [inline README.md]: This link needs to be directed to my repository, like this:
`3. Submit a [pull request](https://github.com/Alperencode/Algorithm-Solutions/pulls) with a detailed explanation of your solution or problem.`

File: README.md [hunk 4]
```diff
@@ -1,9 +1,73 @@
...
+
+## License
 
-#### Note: Files with the '-u' at the end means they are unfinished.
+This project is open-source and generously shared with the community under the [MIT License](https://github.com/finn-str/Algorithm-Solutions/blob/main/LICENSE). Feel free to use, modify, and distribute as you see fit, but do give the rightful credit.
```
Comment (Alperencode) [inline README.md]: This repository does not have a license so this section needs to be **removed**.

File: README.md [hunk 5]
```diff
@@ -1,9 +1,73 @@
...
-<br>
+Note: Files with the '-u' at the end means they are unfinished.
+
+## Table of Contents
+
+- [Features](#features)
+- [Getting Started](#getting-started)
+- [Folder Structure](#folder-structure)
+- [Contribution Guidelines](#contribution-guidelines)
+- [License](#license)
```
Comment (Alperencode) [inline README.md]: This repository does not have a license so this section needs to be **removed**.

**Review by Alperencode — CHANGES_REQUESTED** [review]
Needs to fix:

- Hyperlink URLs
- License section

Other parts looking good, thanks!

**Comment by finn-str** [comment]
I fixed the links and the licence mistake, now you only need to edit the structure.

**Comment by Alperencode** [comment]
@finn-str Looking all good, thanks for the help Finn!

**Pull request merged**